spring-content-rest/src/main/java/org/springframework/data/rest/extensions/entitycontent/RepositoryEntityMultipartController.java

```diff
@@ -130,6 +127,9 @@ public ResponseEntity<RepresentationModel<?>> createEntityAndContent(RootResourc
             }
         }
 
+        AfterCreateEvent afterCreate = new AfterCreateEvent(savedEntity);
+        publisher.publishEvent(afterCreate);
+
```

Comment:
This is not only changing the integration tests if you're moving the place where this event is emitted.

Reply:
This is true.  I am not sure I have a strong opinion in this case.  I cant thing of a reason why the publish cant be at the end of the call, rather than just after the event.  I dont think it materially effects the behavior.  WDYT?

Reply:
The difference is in the ordering of events.

Previously, the `AfterCreateEvent` is emitted before the set content events (which I believe are emitted internally inside the `ContentStoreContentService` methods).
Now, the `AfterCreateEvent` is emitted after the set content events.

Reply:
Right.  I would agree.  But I doubt many folks are using this controller yet besides you folks.  But that is valid.  The only reason for this change is because of my general desire to support multi-threaded tests but we lapse in a bunch of places so if you are not comfortable with the event ordering changing then we probably should force it through in a test change.